From penberg@cs.helsinki.fi  Wed Nov 12 13:52:53 2008
From: Pekka Enberg <penberg@cs.helsinki.fi>
Date: Thu, 30 Oct 2008 18:12:04 +0200
Subject: Staging: w35und: move global wbsoft_enabled to struct wbsoft_priv
To: greg@kroah.com
Cc: Pekka Enberg <penberg@cs.helsinki.fi>, Pavel Machek <pavel@suse.cz>
Message-ID: <1225383127-16406-4-git-send-email-penberg@cs.helsinki.fi>


This is a preparational step for moving packet_came() to wb35rx.c().

Acked-by: Pavel Machek <pavel@suse.cz>
Signed-off-by: Pekka Enberg <penberg@cs.helsinki.fi>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/winbond/core.h  |    2 ++
 drivers/staging/winbond/wbusb.c |   11 ++++++-----
 2 files changed, 8 insertions(+), 5 deletions(-)

--- a/drivers/staging/winbond/core.h
+++ b/drivers/staging/winbond/core.h
@@ -39,6 +39,8 @@ struct wbsoft_priv {
 	struct iw_statistics iw_stats;
 
 	u8 LinkName[WB_MAX_LINK_NAME_LEN];
+
+	bool enabled;
 };
 
 #endif /* __WINBOND_CORE_H */
--- a/drivers/staging/winbond/wbusb.c
+++ b/drivers/staging/winbond/wbusb.c
@@ -46,8 +46,6 @@ static struct ieee80211_supported_band w
 	.n_bitrates	= ARRAY_SIZE(wbsoft_rates),
 };
 
-int wbsoft_enabled;
-
 static int wbsoft_add_interface(struct ieee80211_hw *dev,
 				 struct ieee80211_if_init_conf *conf)
 {
@@ -129,8 +127,10 @@ static int wbsoft_tx(struct ieee80211_hw
 
 static int wbsoft_start(struct ieee80211_hw *dev)
 {
-	wbsoft_enabled = 1;
-	printk("wbsoft_start called\n");
+	struct wbsoft_priv *priv = dev->priv;
+
+	priv->enabled = true;
+
 	return 0;
 }
 
@@ -393,10 +393,11 @@ error:
 
 void packet_came(struct ieee80211_hw *hw, char *pRxBufferAddress, int PacketSize)
 {
+	struct wbsoft_priv *priv = hw->priv;
 	struct sk_buff *skb;
 	struct ieee80211_rx_status rx_status = {0};
 
-	if (!wbsoft_enabled)
+	if (!priv->enabled)
 		return;
 
 	skb = dev_alloc_skb(PacketSize);
